config/base.yml

```diff
@@ -165,13 +165,13 @@ before_script:
 
   # codecov
   # https://about.codecov.io/blog/introducing-codecovs-new-uploader/
-  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --import; fi
-  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/codecov-linux; fi
-  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/codecov-linux.SHA256SUM; fi
-  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/codecov-linux.SHA256SUM.sig; fi
-  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then gpg --verify codecov-linux.SHA256SUM.sig codecov-linux.SHA256SUM; fi
-  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then shasum -a 256 -c codecov-linux.SHA256SUM; fi
-  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then chmod +x codecov-linux; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/linux/codecov; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM.sig ; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then gpgv codecov.SHA256SUM.sig codecov.SHA256SUM; fi
```

Comment:
Do we know for a fact that this will stop execution if it fails?

Reply:
:thinking: I actually don't know

I tried modifying codecov.SHA256SUM on my local to see what happens when it's invalid.  Though I don't know if that would cause travis to hard fail
```
$ gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
gpgv: Signature made Sun 06 Feb 2022 09:40:51 NZDT
gpgv:                using RSA key 27034E7FDB850E0BBC2C62FF806BB28AED779869
gpgv: Good signature from "Codecov Uploader (Codecov Uploader Verification Key) <security@codecov.io>"
$ shasum -a 256 -c codecov.SHA256SUM
codecov: OK
$ nano codecov.SHA256SUM
$ shasum -a 256 -c codecov.SHA256SUM
shasum: codecov.SHA256SUM: no properly formatted SHA1 checksum lines found
$ gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
gpgv: Signature made Sun 06 Feb 2022 09:40:51 NZDT
gpgv:                using RSA key 27034E7FDB850E0BBC2C62FF806BB28AED779869
gpgv: BAD signature from "Codecov Uploader (Codecov Uploader Verification Key) <security@codecov.io>"
```

Reply:
I've had problem in the past where I was expecting GH Actions to stop when something went wrong, but it kept going. 

From memory, I think Travis will stop on any `before_script' command that returns an error code but I'm not sure.